bayesian/utils.py

```diff
@@ -207,16 +207,19 @@ def convert_version_to_proper_semantic(version, package_name=None):
     try:
         if version in ('', '-1', None):
             version = '0.0.0'
-        """Needed for maven version like 1.5.2.RELEASE to be converted to
-        1.5.2 - RELEASE for semantic version to work."""
-        version = version.replace('.', '-', 3)
-        version = version.replace('-', '.', 2)
-        # Needed to add this so that -RELEASE is account as a Version.build
-        version = version.replace('-', '+', 3)
+
+        if ecosystem == "maven":
+            """Needed for maven version like 1.5.2.RELEASE to be converted to
+            1.5.2 - RELEASE for semantic version to work."""
+            version = version.replace('.', '-', 3)
```

Comment:
what value will be sent for ingestion in case the pkg is unknown? Also, is the version in graph going to be stored with . or - ? While fetching details from graph, will it expect in . or - form?

Reply:
This functions is used to format the version we report in recommended version field of CA POST call. This is not used for any query graph / ingestion purpose. Also I don't know why `maven` version are converted from '.' to '-' forms, this generic conversion was creating issue for golang for sure, so moved maven specific code under ecosystem check.

Reply:
ok. can you ask the original coder to check why do we need this in the first place?

Reply:
@deepak1725 Can you add oldie who can provide us reasoning behind this strange version transformation for Maven ecosystem?

Reply:
This code was written 3 years ago, You will hardly found the author of that.